include/merlin/core_kernels/upsert_and_evict.cuh

```diff
@@ -210,9 +211,10 @@ __global__ void tlp_v1_upsert_and_evict_kernel_unique(
             verify_score_ptr->load(cuda::std::memory_order_relaxed);
         if (verify_score <= min_score) {
           key_pos = min_pos;
-          score = scores == nullptr ? device_nano<S>() : score;
-          update_score_digest<K, V, S>(bucket_keys_ptr, bucket_capacity,
-                                       key_pos, key, score);
+          ScoreFunctor::update_with_digest(
+              bucket_keys_ptr, key_pos, scores, kv_idx, score, bucket_capacity,
+              get_digest<K>(key), (occupy_result != OccupyResult::DUPLICATE));
```

Comment:
Actually, occupy_result is not updated here, its value is INITIAL.
Anyway, here, occupy_result != OccupyResult:DUPLICATE always true.

Reply:
> Actually, occupy_result is not updated here, its value is INITIAL. Anyway, here, occupy_result != OccupyResult:DUPLICATE always true.

If here means the key is missed, this flag can be set to be `True`. Anyway, the current logic looks right.